wagtail/wagtailcore/query.py

```diff
@@ -350,6 +350,12 @@ def specific(self):
         else:
             return self._clone(klass=SpecificQuerySet)
 
+    def in_site(self, site):
+        """
+        This filters the QuerySet to only contain pages of type self in a specific site.
```

Comment:
I don't think "pages of type self" is accurate - there's no reason to assume that the queryset has been (or will be) filtered by page type. Perhaps go for "This filters the QuerySet to only contain pages within the specified site"?